src/matchers/excel.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const { OneDrive } = require('@adobe/helix-onedrive-support');
+
+async function extract(url, params, log = console) {
+  const {
+    AZURE_WORD2MD_CLIENT_ID: clientId,
+    AZURE_HELIX_USER: username,
+    AZURE_HELIX_PASSWORD: password,
+  } = params;
+
+  try {
+    const drive = new OneDrive({
+      clientId,
+      username,
+      password,
+      log,
+    });
+
+    const item = await drive.getDriveItemFromShareLink(url);
+    const worksheetsuri = `/drives/${item.parentReference.driveId}/items/${item.id}/workbook/worksheets/`;
+    const worksheets = await (await drive.getClient(false)).get(worksheetsuri);
```

Comment:
- I think you can reuse the client for all requests
- the `raw` argument is `false` by default. so `const client = await drive.getClient()` should be enough.